main.cpp

```diff
@@ -384,96 +318,164 @@ each property for UDTs 5 - 9.
+    3) whether cup full (bool)
+    4) type of coffee (espesso/cappuccino/etc)
+    5) amount of water remaining in reservoir
 3 things it can do:
-    1)
-    2)
-    3)
+    1) heat water
+    2) deliver correct quantity of heated water
+    3) beep when pouring complete
  */
 
 /*
-Thing 6)
+Thing 2) Dentist
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) number of patients per day
+    2) maximum number of patients per week
+    3) disposable equipment costs per week
+    4) profit per week
+    5) number of staff
 3 things it can do:
-    1)
-    2)
-    3)
+    1) treat patient
+    2) charge patient fee for service
+    3) pay staff wages
  */
 
 /*
-Thing 7)
+Thing 3) Automatic public toilet
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) volume of cistern
+    2) volume of cleaning solution reservoir
+    3) number of uses per day
+    4) brightness of lights
+    5) amount of water per flush
 3 things it can do:
-    1)
-    2)
-    3)
+    1) flush
+    2) illuminate
+    3) clean
  */
 
 /*
-Thing 8)
+Thing 4) ATM
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) current amount in machine in dollars
+    2) maximum amount machine can hold in dollars
+    3) number of customers per day
+    4) customer account number
+    5) customer pin number
 3 things it can do:
-    1)
-    2)
-    3)
+    1) dispense cash
+    2) display balance
+    3) dispense receipt
  */
 
 /*
-Thing 9)
+Thing 5) Cabin
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) Height in cm (float)
+    2) Depth in cm (float)
+    3) Width in cm (float)
+    4) Wall colour (std::string)
+    5) Capacity in persons (int)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) set size
+    2) set weight
+    3) set capacity
  */
 
 /*
-===================
-Part 1d assignment: Step 3
-===================
-You've just defined 10 UDTs!
-4 of them are unrelated (UDTs 1-4).
-5 of them form the sub-objects that make up the 10th UDT. 
-
-MOVE THEM BELOW this block comment and put them in numerical order 
-(1 - 10) starting on line 473
-    DO NOT COPY. 
-    I do not want to see 2 copies of your Plain-English UDTs.  
-    I only want to see the 10 UDTs written BELOW this block comment, in numerical order (1 - 10).
+Thing 6) Door
+5 properties:
+    1) Height in cm (float)
+    2) Width in cm (float)
+    3) Speed of movement (float)
+    4) Colour (std::string)
+    5) Status open vs closed (bool)
+3 things it can do:
+    1) Open
+    2) Close 
+    3) Get status (open or closed)
+ */
 
-After you have MOVED your 10 UDTs, send me a DM with your pull request link.
-I will review the pseudo-code that you have written.
-*/
+/*
+Thing 7) Panel
+5 properties:
+    1) Height in cm (int)
+    2) Width in cm (int)
+    3) Number of buttons (int)
+    4) Size of buttons (int)
+    5) Brightness of backlighting (int)
+3 things it can do:
+    1) Choose panel size
+    2) Choose number of buttons
+    3) Change size of buttons
```

Comment:
These 3 objects all have `Height` and `Width` properties.
That's not very creative and seems like a copy-and-paste approach to the assignment. 

Please see my message in Slack
- [ ] optionally revise, pending slack message decision